File: app/src/main/java/net/tlalka/android/fiszki/model/dao/LessonDao.java
package net.tlalka.android.fiszki.model.dao;

import com.j256.ormlite.support.ConnectionSource;
import net.tlalka.android.fiszki.model.entities.Lesson;
import net.tlalka.android.fiszki.model.types.LanguageType;

import java.sql.SQLException;
import java.util.List;

public class LessonDao extends AbstractDao<Lesson, Long> {

    public LessonDao(ConnectionSource connectionSource) throws SQLException {
        super(connectionSource, Lesson.class);
    }

    public Lesson getLessonBy(long id) throws SQLException {
        return super.queryForId(id);
    }

    public List<Lesson> getLessonsBy(LanguageType languageType) throws SQLException {
        return super.queryBuilder()
                .orderBy("ownerType", true)
                .orderBy("levelType", true)
                .where()
                .eq("languageType", languageType)
                .query();
    }
}

File: app/src/main/kotlin/net/tlalka/fiszki/model/dao/LessonDao.kt
package net.tlalka.fiszki.model.dao

import com.j256.ormlite.support.ConnectionSource
import net.tlalka.fiszki.model.entities.Lesson
import net.tlalka.fiszki.model.types.LanguageType
import java.sql.SQLException

class LessonDao @Throws(SQLException::class)
constructor(connectionSource: ConnectionSource) : AbstractDao<Lesson, Long>(connectionSource, Lesson::class.java) {

    @Throws(SQLException::class)
    fun getLessonBy(id: Long): Lesson {
        return super.queryForId(id)
    }

    @Throws(SQLException::class)
    fun getLessonsBy(languageType: LanguageType): List<Lesson> {
        return super.queryBuilder()
                .orderBy("ownerType", true)
                .orderBy("levelType", true)
                .where()
                .eq("languageType", languageType)
                .query()
    }
}
